api/src/pcapi/routes/pro/stocks.py

```diff
@@ -189,7 +190,20 @@ def upsert_stocks(
     except offers_exceptions.OfferEditionBaseException as error:
         raise ApiErrors(error.errors, status_code=400)
 
-    offers_api.handle_stocks_edition(edited_stocks_with_update_info)
+    try:
+        offers_api.handle_stocks_edition(edited_stocks_with_update_info)
+    except booking_exceptions.BookingIsAlreadyCancelled:
+        raise ApiErrors({"global": ["Cette contremarque a déjà été annulée"]}, status_code=410)
+    except booking_exceptions.BookingIsAlreadyRefunded:
+        raise ApiErrors({"global": ["Impossible d'annuler une réservation consommée"]}, status_code=410)
+    except booking_exceptions.BookingHasActivationCode:
+        raise ApiErrors(
+            {"global": ["Cette contremarque a des codes d'activation. Sa validation ne peut pas être annulée"]},
+            status_code=403,
+        )
+    except booking_exceptions.BookingIsNotUsed:
+        raise ApiErrors({"global": ["Cette contremarque n'a pas encore été validée"]}, status_code=410)
```

Comment:
[détail] On pourrait faire comme dans le fichier `api/src/pcapi/routes/public/booking_token/v2/endpoints.py` et utiliser `ResourceGoneError` et `ForbiddenError` au lieu de `ApiErrors` et `status_code` : 

```suggestion
        raise ResourceGoneError({"global": ["Cette contremarque a déjà été annulée"]})
    except booking_exceptions.BookingIsAlreadyRefunded:
        raise ResourceGoneError({"global": ["Impossible d'annuler une réservation consommée"]})
    except booking_exceptions.BookingHasActivationCode:
        raise ForbiddenError(
            {"global": ["Cette contremarque a des codes d'activation. Sa validation ne peut pas être annulée"]}
        )
    except booking_exceptions.BookingIsNotUsed:
        raise ResourceGoneError({"global": ["Cette contremarque n'a pas encore été validée"]})
```

Reply:
Ce point a été traité dans le 2ème commit. Il faudrait réintégrer les changements de ce fichier dans le 1er commit. Le 2ème commit ne devrait contenir que des modifications dans `api/src/pcapi/routes/public/individual_offers/v1/*.py`.

Et je remarque aussi que les clés des messages d'erreur ont changé : la clé était `booking` (ou `payment` dans le cas d'une exception `BookingIsAlreadyRefunded`), mais désormais la clé est `global`. Je me demande si ça ne pourrait pas casser quelque chose côté portail pro. En tout cas, je ne pense pas qu'on ait de raison de modifier ces clés. Qu'en penses-tu ?

Reply:
Je suis d'accord, j'ai remis les mêmes clés et messages pour les anciennes routes, pour la nouvelle route crée, j'ai mis des message que je trouve plus explicite